pygac/calibration/noaa.py

```diff
@@ -40,14 +40,80 @@
 LOG = logging.getLogger(__name__)
 
 
+def calibrate(ds, custom_coeffs=None, coeffs_file=None):
+    """Calibrate the dataset `ds`, possibly using custom_coeffs or a coeff file.
+
+    Args:
+        ds: xarray Dataset object containing the data to calibrate.
+        custom_calibration: dictionary with a subset of user defined satellite specific
+                            calibration coefficients
+        calibration_file: path to json file containing default calibrations
+    """
+    channels = ds["channels"].data
+    times = ds.coords["times"].data
+    scan_line_numbers = ds["scan_line_index"].data
+
+    calibration_coeffs = Calibrator(
+            ds.attrs["spacecraft_name"],
+            custom_coeffs=custom_coeffs,
+            coeffs_file=coeffs_file
+        )
+
+    # `times` is in nanosecond resolution. However, convertion from datetime64[ns] to datetime objects
+    # does not work, and anyway only the date is needed.
+    start_time = times[0]
+    start_date = start_time.astype("datetime64[D]").item()
+    year = start_date.year
+
+    delta = (start_time.astype("datetime64[D]") - start_time.astype("datetime64[Y]")).astype(int)
+    jday = delta.astype(int) + 1
```

Comment:
Can this be simplified using a datetime accessor? For example

```
t = ds.coords["times"]
jday = t.dt.dayofyear
year = t.dt.year
```

Reply:
Cool, I didn't know you could do that! fixed